src/compile/axis/config.ts

```diff
@@ -1,20 +1,56 @@
-import {ScaleType} from 'vega-typings/types';
-import {array} from 'vega-util';
+import {ScaleType, SignalRef} from 'vega-typings/types';
+import {array, stringValue} from 'vega-util';
 import {AxisConfig} from '../../axis';
 import {PositionScaleChannel} from '../../channel';
 import {Config} from '../../config';
 import {isQuantitative} from '../../scale';
-import {titlecase} from '../../util';
+import {keys, titlecase} from '../../util';
+import {isSignalRef} from '../../vega.schema';
 import {getStyleConfig} from '../common';
 
-function getAxisConfigFromConfigTypes(configTypes: string[], config: Config) {
+function getAxisConfigFromConfigTypes(
+  configTypes: string[],
+  config: Config,
+  channel: 'x' | 'y',
+  orient: string | SignalRef
+) {
   // TODO: add special casing to add conditional value based on orient signal
-  return Object.assign.apply(null, [{}, ...configTypes.map(configType => config[configType])]);
+  return Object.assign.apply(null, [
```

Comment:
Use `...` instead.

Reply:
I already use ... below, so this comment doesn't make sense.

Plus, you can't do ... to merge/assign an array of object.

Reply:
We could if we do it in a for loop. Anyway, not needed.